assignments/assignment01.rb

```diff
@@ -38,7 +38,26 @@ def number_to_string(n, lang)
 # it accepts a string
 # and returns the same string with each word capitalized.
 def titleize(s)
-  # your implementation here
+  return 'No string included.  Do it like this: maketitle "your string"' if s == ""
+
+  chars = s.chars
+
+  result = chars[0].capitalize
+
+  counter = 1  
+  while counter < s.length
+    c = chars[counter]
+    p = chars[counter - 1]
+    if (p == " ") & (c != " ")
+      result << c.capitalize
+      counter = counter + 1
+    else
+      result << c.downcase
+      counter = counter + 1
+    end
+  end
+
+  result
```

Comment:
Your solution is very un-ruby like and way more complicated than needed. All you need to do is use split on the string to split it into separate words, capitalize each word, and then join the words back together.